integrationTests/node/heartbeatV2/heartbeatV2_test.go

```diff
@@ -0,0 +1,473 @@
+
+	delayedNodeCache := delayedNodeDataPool.PeerAuthentications()
+	assert.Equal(t, len(nodes)-1, delayedNodeCache.Len())
+
+	// Only search for messages from initially created nodes.
+	// Last one does not send peerAuthentication yet
+	for i := 0; i < len(nodes)-1; i++ {
+		assert.True(t, delayedNodeCache.Has(nodes[i].ID().Bytes()))
+	}
+
+	// Create sender for last node
+	nodeName := fmt.Sprintf("%s%d", defaultNodeName, newNodeIndex)
+	sk, _ := keyGen.GeneratePair()
+	s := createSender(nodeName, delayedNode, sigHandler, sk)
+	senders = append(senders, s)
+
+	// Wait to make sure all peers send messages again
+	time.Sleep(time.Second * 3)
+
+	// Check sent messages again - now should have from all peers
+	maxMessageAgeAllowed = time.Second * 5 // should not have messages from first Send
+	checkMessages(t, nodes, dataPools, maxMessageAgeAllowed)
+
+	// Add new delayed node which requests messages by hash array
+	delayedNode, delayedNodeDataPool = createDelayedNode(nodes, sigHandler)
+	nodes = append(nodes, delayedNode)
+	dataPools = append(dataPools, delayedNodeDataPool)
+	delayedNodeResolver := createPeerAuthResolver(pksArray, delayedNodeDataPool.PeerAuthentications(), delayedNode, shardCoordinator)
+	_ = delayedNodeResolver.RequestDataFromHashArray(pksArray, 0)
+
+	// Wait for messages to broadcast
+	time.Sleep(time.Second * 3)
+
+	// Check that the node received peer auths from all of them
+	assert.Equal(t, len(nodes)-1, delayedNodeDataPool.PeerAuthentications().Len())
+	for _, node := range nodes {
+		assert.True(t, delayedNodeDataPool.PeerAuthentications().Has(node.ID().Bytes()))
+	}
+
+	closeComponents(t, nodes, senders, dataPools, paResolvers)
+}
+
+func TestHeartbeatV2_NetworkShouldSendMessages(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	keyGen := signing.NewKeyGenerator(mcl.NewSuiteBLS12())
+	sigHandler := createMockPeerSignatureHandler(keyGen)
+
+	nodes, _ := integrationTests.CreateFixedNetworkOf8Peers()
+	interactingNodes := len(nodes)
+
+	// Create components
+	dataPools := make([]dataRetrieverInterface.PoolsHolder, interactingNodes)
+	senders := make([]factory.HeartbeatV2Sender, interactingNodes)
+	for i := 0; i < interactingNodes; i++ {
+		dataPools[i] = dataRetriever.NewPoolsHolderMock()
+		createPeerAuthMultiDataInterceptor(nodes[i], dataPools[i].PeerAuthentications(), sigHandler)
+		createHeartbeatMultiDataInterceptor(nodes[i], dataPools[i].Heartbeats(), sigHandler)
+
+		nodeName := fmt.Sprintf("%s%d", defaultNodeName, i)
+		sk, _ := keyGen.GeneratePair()
+
+		s := createSender(nodeName, nodes[i], sigHandler, sk)
+		senders[i] = s
+	}
+
+	// Wait for all peers to send peer auth messages twice
+	time.Sleep(time.Second * 15)
+
+	checkMessages(t, nodes, dataPools, time.Second*7)
+
+	closeComponents(t, nodes, senders, dataPools, nil)
+}
+
+func createDelayedNode(nodes []p2p.Messenger, sigHandler crypto.PeerSignatureHandler) (p2p.Messenger, dataRetrieverInterface.PoolsHolder) {
```

Comment:
in order to ease the setup we could have a specialized test node that could contain the p2p messenger, heartbeat components, some caches, the resolvers & interceptors.
In this way this specialized node could have been reused elsewhere and would contain only the required components.

Reply:
done